shellai/config.py

```diff
@@ -0,0 +1,132 @@
+import json
+import logging
+import os
+import sys
+from collections import namedtuple
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+CONFIG_DEFAULT_PATH: str = "~/.config/shellai/config.toml"
+
+# tomllib does not support writting files, so we will create our own.
+CONFIG_TEMPLATE = """
+[output]
+# otherwise recording via script session will be enforced
+enforce_script = {enforce_script}
+# file with output(s) of regular commands (e.g. ls, echo, etc.)
+file = "{output_file}"
+# Keep non-empty if your file contains only output of commands (not prompt itself)
+prompt_separator = "{prompt_separator}"
+
+[history]
+enabled = {enabled}
+file = "{history_file}"
+# max number of queries in history (including responses)
+max_size = {max_size}
+
+[backend]
+endpoint = "{endpoint}"
+
+"""
+
+
+class Output(namedtuple("Output", ["enforce_script", "file", "prompt_separator"])):
```

Comment:
Have you considered using `dataclasses` instead of `namedtuple`? It might look something like this, though I'm not sure if you would still need to override `__new__()` to get the desired behavior.
```python
@dataclasses.dataclass(slots=True)
class Output:
    enforce_script: bool = false
    file: str = "/tmp/shellai_output.txt"
    prompt_separator: str = "$"
```

Reply:
If we manage to get rid of RHEL 8, then I would be totally fine with using dataclasses

Reply:
Sounds good. I forgot about needing to support Python 3.6 since I switched all my RHEL 8 machines to 3.9 in the app stream.

Reply:
Our `pyproject.toml` says we require Python 3.9. Maybe we need to adjust that if we do actually support older Python versions.